lib/markdownlint.js

```diff
@@ -895,6 +895,38 @@ function parseConfiguration(name, content, parsers) {
   };
 }
 
+/**
+ * Resolve referenced "extends" path in a configuration file
+ * using path.resolve() and require.resolve() as a fallback.
+ *
+ * @param {string} configFile Configuration file name.
+ * @param {string} referenceId Referenced identifier to resolve.
+ * @returns {string} Resolved path to file
+ */
+function resolveConfigExtends(configFile, referenceId) {
+  const configFileDirname = path.dirname(configFile);
+  const resolvedExtendsFile = path.resolve(configFileDirname, referenceId);
+
+  try {
+    if (fs.statSync(resolvedExtendsFile).isFile()) {
+      return resolvedExtendsFile;
+    }
+  // eslint-disable-next-line unicorn/prefer-optional-catch-binding
+  } catch (error) {
+    // If fs.statSync has thrown, trying require.resolve
+  }
+
+  try {
+    return require.resolve(referenceId, { "paths": [ configFileDirname ] });
+
+  // eslint-disable-next-line unicorn/prefer-optional-catch-binding
+  } catch (error) {
+    // If require.resolve throws, returning resolvedExtendsFile for BC
+  }
+
```

Comment:
Nit: I think in the rest of the file there are not blank lines between sections like this.